tests/test_filter_advisories.py

```diff
@@ -0,0 +1,240 @@
+                },
+                {
+                    "vulnerable_version_range": "= 0.11.1",
+                    "patched_versions": "",
+                },
+            ],
+        },
+    ]
+
+
+class TestFilterAdvisories:
+
+    @pytest.mark.parametrize(
+        "minimal_version,last_run_date,expected_count,expected_ids",
+        [
+            (
+                "0.11.0",
+                "2023-05-31T00:00:00+00:00",
+                2,
+                ["GHSA-xxxx-xxxx-xxxx", "GHSA-yyyy-yyyy-yyyy"],
+            ),
+            (
+                "0.12.0",
+                "2023-05-31T00:00:00+00:00",
+                1,
+                ["GHSA-yyyy-yyyy-yyyy"],
+            ),
+            (
+                "0.11.1",
+                "2023-05-31T00:00:00+00:00",
+                3,
+                ["GHSA-xxxx-xxxx-xxxx", "GHSA-yyyy-yyyy-yyyy", "GHSA-zzzz-zzzz-zzzz"],
+            ),
+            (
+                "0.11.1",
+                "2023-07-02T00:00:00+00:00",
+                1,
+                ["GHSA-zzzz-zzzz-zzzz"],
+            ),
+        ],
+    )
+    def test_filter_advisories(
+        self, advisories, minimal_version, last_run_date, expected_count, expected_ids
+    ):
+        minimal_version = semantic_version.Version(minimal_version)
+        last_run_date = datetime.fromisoformat(last_run_date)
+
+        filtered = filter_advisories(advisories, minimal_version, last_run_date)
+        assert len(filtered) == expected_count
+        assert [adv["ghsa_id"] for adv in filtered] == expected_ids
+
+    def test_filter_advisories_validation_error(self):
+        minimal_version = semantic_version.Version("0.11.0")
+        last_run_date = datetime.fromisoformat("2023-06-01T00:00:00+00:00")
+        broken_advisories = [{"ghsa_id": "GHSA-zzzz-zzzz-zzzz", "published_at": ""}]
+
+        with pytest.raises(ValidationError):
+            filter_advisories(broken_advisories, minimal_version, last_run_date)
+
+
+class TestParseVulnerabilities:
+
+    @pytest.mark.parametrize(
+        "minimal_version,advisory_id,expected_count,expected_ids",
+        [
+            ("0.11.0", "GHSA-xxxx-xxxx-xxxx", 1, ["GHSA-xxxx-xxxx-xxxx"]),
+            ("0.12.0", "GHSA-xxxx-xxxx-xxxx", 0, []),
+            ("0.11.0", "GHSA-yyyy-yyyy-yyyy", 1, ["GHSA-yyyy-yyyy-yyyy"]),
+            ("0.11.0", "GHSA-zzzz-zzzz-zzzz", 0, []),
+            ("0.11.1", "GHSA-zzzz-zzzz-zzzz", 1, ["GHSA-zzzz-zzzz-zzzz"]),
+            ("0.10.0", "GHSA-zzzz-zzzz-zzzz", 1, ["GHSA-zzzz-zzzz-zzzz"]),
+        ],
+    )
+    def test_parse_vulnerabilities(
+        self, advisories, minimal_version, advisory_id, expected_count, expected_ids
+    ):
+        minimal_version = semantic_version.Version(minimal_version)
+        filtered = []
+
+        advisory = next(adv for adv in advisories if adv["ghsa_id"] == advisory_id)
+        filtered = parse_vulnerabilities(advisory, filtered, minimal_version)
+
+        assert len(filtered) == expected_count
```

Comment:
I think this would be equivalent to `len(expected_ids)`, and you could drop the expected count param.